sds_data_manager/lambda_code/SDSCode/pipeline_lambdas/batch_starter.py

```diff
@@ -396,6 +397,11 @@ def try_to_submit_job(
     serialized_dependencies : str
         The serialized ProcessingInputCollection of the upstream
         dependencies.
+    skip_duplicate_jobs : bool
+        In normal reprocessing (due to a new file upload) duplicate jobs should be
+        skipped to avoid redundant products. In cases where a user has manually
+        triggered reprocessing, we want to reprocess even if the job looks like a
+        duplicate. Default is True.
```

Comment:
I liked this point you mentioned.
```suggestion
        duplicate. Eg. If algorithm got updated but the input remained same. Default is True.
```